source/Audio.cpp

```diff
@@ -256,10 +264,13 @@ void Audio::Play(const Sound *sound, const Point &position)
 	if(!isInitialized || !sound || !sound->Buffer() || !volume)
 		return;
 	
-	// Place sounds from the main thread directly into the queue. They are from
-	// the UI, and the Engine may not be running right now to call Update().
+	// Play sounds from the main thread directly on dedicated UI Source
 	if(this_thread::get_id() == mainThreadID)
-		queue[sound].Add(position - listener);
+	{
+		uiSoundFX.emplace_back(sound, uiSource);
```

Comment:
Do elements actually ever accumulate in the `uiSoundFX` vector? it seems whenever any would be added, they are then removed. Does it make sense to instead simply construct the source here, e.g.
```c++
{
  auto toPlay = Source{sound, uiSource};
  alSourcePlay(uiSource);
}
```
as it will be automatically cleaned up when this block scope exits.

Or, is the vector needed, as the Source needs to exist for some time, in which case the call to `pop_back` needs to be elsewhere?

Reply:
Good point. Further, do we need the toPlay at all? The return isn't used. We seem to be looking for the side effect of modifying the uiSource passed in to Source.

I made it like this:
`	{
		Source{sound, uiSource};
		alSourcePlay(uiSource);
	}`

and it still appears to work. However, I'm not sure if that is bad for some reason.

Reply:
It is indeed bad to create an object that immediately goes out-of-scope.

We must guarantee that the `Source` remains valid while OpenAl needs it. Whether that is accomplished by storing it in a vector and then removing it elsewhere, or whether it living for this block scope only is sufficient, is something that you will need to determine.

Reply:
Ok, I think I figured it out. I am now creating the source and using that ID to play.